### ISSUE TITLE
Expo-camera isn't functioning on my MacBook, but works on my Windows PC (web only)

### ISSUE THREAD

**Issue by Sub-Zero-1** [expo/expo#28337]
### Minimal reproducible example

https://snack.expo.dev/@bycedric/expo-issue-15442

### What platform(s) does this occur on?

Web

### Did you reproduce this issue in a development build?

No (web-only issue)

### Summary

I'm attempting to implement a QR scanner across Android, iOS, and web platforms. While it functions correctly on Android and iOS, I'm encountering issues on the web. Specifically, the camera initializes, but it fails to recognize any QR codes. Despite trying various solutions, the problem persists.

Code Snippet:

```
<CameraView
  onBarcodeScanned={handleBarCodeScanned}
  barcodeScannerSettings={{ barcodeTypes: ["qr"] }}
/>
```

Additional Details:

I came across another code example that has the same problem: https://snack.expo.dev/@bycedric/expo-issue-15442. Upon testing this example on my MacBook using both Chrome and Safari, I encountered the same issue - no QR codes were recognized. Intriguingly, when I tried the example on a Windows PC using Chrome, it worked flawlessly.

Question:

What could be causing this inconsistent behavior across different platforms and browsers? Any insights or suggestions would be greatly appreciated!

### Environment

expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.3
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 18.12.1 - /usr/local/bin/node
      Yarn: 1.22.19 - /opt/homebrew/bin/yarn
      npm: 8.19.2 - /usr/local/bin/npm
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.4, iOS 17.4, macOS 14.4, tvOS 17.4, visionOS 1.1, watchOS 10.4
    IDEs:
      Android Studio: 2022.3 AI-223.8836.35.2231.11005911
      Xcode: 15.3/15E204a - /usr/bin/xcodebuild
    npmPackages:
      expo: ~50.0.14 => 50.0.15 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.73.6 => 0.73.6 
      react-native-web: ~0.19.6 => 0.19.10 
    npmGlobalPackages:
      expo-cli: 6.3.10
    Expo Workflow: managed

### Expo Doctor Diagnostics

npx expo-doctor         
✔ Check Expo config for common issues
✔ Check package.json for common issues
✔ Check native tooling versions
✔ Check dependencies for packages that should not be installed directly
✔ Check for common project setup issues
✔ Check npm/ yarn versions
✔ Check for issues with metro config
✔ Check Expo config (app.json/ app.config.js) schema
✔ Check that packages match versions required by installed Expo SDK
✔ Check that native modules do not use incompatible support packages
✔ Check for legacy global CLI installed locally
✔ Check that native modules use compatible support package versions for installed Expo SDK

Didn't find any issues with the project!

**Comment by byCedric** [comment]
Hi @Sub-Zero-1! I'm the one who created that Snack, but I can't seem to reproduce this on my MacBook using either Chrome or Safari, both work fine on my end.

<details><summary>Tested QR code and results</summary>

QR Code | Results
--- | ---
![First QR code result on google](https://github.com/expo/expo/assets/1203991/bb3de35f-7967-4e15-b994-282fd58eede5) | ![Snack logs](https://github.com/expo/expo/assets/1203991/d5a65056-1456-40f9-88f6-cb7a04ee6174)

</details>

I'd be happy to dive into the issue once we can reproduce this. If you have another example that might be triggering this issue, please do send it.

**Comment by Sub-Zero-1** [comment]
Hi @byCedric,

First, I scanned your specific QR code with your Snack, and it worked once. Then it stopped working again. I restarted my MacBook and turned off the camera permission for Chrome, then turned it back on. After that, it worked again. I'm now using your Snack code in my app, but I had to downgrade expo-camera and expo-barcode-scanner and expo itself to the versions in your code, and it's working as well. Thanks! It required a restart and a change in permissions.
